x-pack/plugins/security/public/analytics/register_user_context.ts

```diff
@@ -12,6 +12,11 @@ import { Sha256 } from '@kbn/crypto-browser';
 
 import type { AuthenticationServiceSetup } from '..';
 
+interface UserIdContext {
+  userId?: string;
+  isElasticCloudUser: boolean;
+}
```

Comment:
We need to explicitly declare the type because, otherwise, it infers the type returned by `RxJS.map` (where `userId` is a string). Then the schema fails to compile because it cannot be declared as optional.

Declaring the type with `userId` as explicitly optional solves this :)